### PR TITLE
Update remark and linter

### PR DESCRIPTION
**Pull request by wooorm**
Finally! 🎉

This updates all the remark stuff, plus supports linter@2.

Note: users previously did config files like this:

```js
{
  "presets": [
    "lint-consistent"
  ]
}
```

...with this merged, it should instead be like this:

```js
{
  "plugins": [
    "preset-lint-consistent"
  ]
}
```

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -33,14 +39,31 @@ function lint(editor) {
     processor = require('remark');
   }
 
+  if (recommendedWithoutConfig) {
```
Comment (Arcanemagus) [inline lib/index.js]: After doing this on another plugin that has many dependencies to check loading of, I would recommend moving the loading of these (and `engine` + `processor`) into a function, still with all the checks for already being loaded. Then call that function here and in the idle callback over in `activate`.

This way Atom will opportunistically load them during an idle period after activation, but still loads them immediately on first `lint()` call if they haven't already been loaded yet.

File: spec/linter-markdown-spec.js [hunk 2]
```diff
@@ -30,14 +30,11 @@ describe('The remark-lint provider for Linter', () => {
...
       const messages = await lint.provideLinter().lint(editor);
-      expect(messages[0].type).toEqual('Warning');
-      expect(messages[0].text).not.toBeDefined();
-      expect(messages[0].html).toEqual(
-        '<span class="badge badge-flexible">remark-lint:' +
-        'no-unused-definitions</span> Found unused definition'
-      );
-      expect(messages[0].filePath).toMatch(/.+definition-use-invalid\.md$/);
-      expect(messages[0].range).toEqual([[2, 0], [2, 58]]);
+      expect(messages[0].severity).toEqual('warning');
```
Comment (Arcanemagus) [inline spec/linter-markdown-spec.js]: Move to `toBe` here (`===` vs `==`).

File: spec/linter-markdown-spec.js [hunk 3]
```diff
@@ -30,14 +30,11 @@ describe('The remark-lint provider for Linter', () => {
...
+      expect(messages[0].location.file).toMatch(/.+definition-use-invalid\.md$/);
```
Comment (Arcanemagus) [inline spec/linter-markdown-spec.js]: Same thing, `toBe` instead of `toEqual` (assuming `remark` outputs OS specific paths).

File: lib/index.js [hunk 4]
```diff
@@ -12,19 +12,25 @@
 import { CompositeDisposable } from 'atom';
 
 const path = require('path');
```
Comment (Arcanemagus) [inline lib/index.js]: This should be removed as it is no longer needed.

File: lib/index.js [hunk 5]
```diff
@@ -33,14 +39,31 @@ function lint(editor) {
...
+
+  if (consistentWithoutConfig) {
+    if (!consistent) {
+      consistent = require('remark-preset-lint-consistent');
+    }
+    plugins.push(consistent);
+  }
+
+  if (consistentWithoutConfig || recommendedWithoutConfig) {
+    defaultConfig = {plugins: plugins};
```
Comment (Arcanemagus) [inline lib/index.js]: `{plugins: plugins}` should be `{ plugins }` 😉

File: lib/index.js [hunk 6]
```diff
@@ -12,19 +12,25 @@
...
+let recommended;
+let consistent;
 const idleCallbacks = new Set();
 
 // Settings
 let scopes;
 let detectIgnore;
 
 function lint(editor) {
+  let plugins = [];
```
Comment (Arcanemagus) [inline lib/index.js]: This should be `const` not `let`.

**Comment by Arcanemagus** [comment]
I'll take care of the above comments in a minute or two 😉.

**Comment by Arcanemagus** [comment]
Anything else that needs to be in here @wooorm?

**Comment by wooorm** [comment]
Thanks for that! Looks good, nothing left to do except merging & releasing as a major! 🎉

**Comment by Arcanemagus** [comment]
I can take care of that now unless you want to do the honors 😉.

**Comment by wooorm** [comment]
No thanks, go ahead!
Sorry again for taking so long, and thanks for the cleanup and other PRs!

**Pull request merged**